Core.ApplicationServices/References/ReferenceService.cs

```diff
@@ -182,16 +182,16 @@ private void RaiseRootUpdated(IEntityWithExternalReferences owner)
                 .GetRootEntity(rootId, rootType)
                 .Match(root =>
                 {
-                    //Clear existing state
-                    root.ClearMasterReference();
-                    var deleteResult = DeleteExternalReferences(root);
-                    if (deleteResult.Failed)
+                    var referenceList = externalReferences.ToList();
+
+                    var referencesToDelete = root.ExternalReferences.Where(externalReference => !referenceList.Any(referenceToUpdate => referenceToUpdate.Uuid == externalReference.Uuid));
```

Comment:
consider this line once again remembering what we talked about in terms of iterations when doing filtering